src/platform/backends/lxd/lxd_mount_handler.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "lxd_mount_handler.h"
+#include "lxd_request.h"
+
+namespace
+{
+constexpr std::string_view category = "lxd-mount-handler";
+constexpr int length_of_unique_id_without_prefix = 25;
+constexpr int timeout_millisecond = 300000;
+} // namespace
+
+namespace multipass
+{
+LXDMountHandler::LXDMountHandler(mp::NetworkAccessManager* network_manager, LXDVirtualMachine* lxd_virtual_machine,
+                                 const SSHKeyProvider* ssh_key_provider, const std::string& target_path,
+                                 const VMMount& mount)
+    : MountHandler{lxd_virtual_machine, ssh_key_provider, target_path, mount.source_path},
+      network_manager_{network_manager},
+      lxd_instance_endpoint_{
+          QString("%1/instances/%2").arg(lxd_socket_url.toString()).arg(lxd_virtual_machine->vm_name.c_str())},
+      // 27 (25 + 2(d_)) letters is the maximal deivce name length that lxd can accepte
+      device_name_{
+          mp::utils::make_uuid(target_path).left(length_of_unique_id_without_prefix).prepend("d_").toStdString()}
```

Comment:
I would like to follow the nomenclature 'devce_<hash>', since 'device' alone took too many digits, so I have shortened it to 'd_', the intention is to make it indicative that it is a device name.

Reply:
I see... 'd_' is a little cryptic though, I am not sure it helps the reader at all. However, LXD might need the first character to be alphabetical. That problem was already solved for the QEMU case, I would think an analogous approach would be best here.

Reply:
Right. With the comment there explaining that the first char needs to be alphanumerical, one quickly understands why the `m` is there, and that it must stand for "mount". 

I would have liked to see a consistent approach in both cases, so that I don't have to re-understand one if I already understood the other. Consistency is definitely not the most important value, but it is not valueless either.

If the two cases were equivalent, you could also change the QEMU case. However, in this respect I find that the QEMU implementation is slightly better because:

1. it explains why something is prepended at all; 
2. it is obvious what the prefix stands for;
3. this looses an extra UUID digit to have the underscore (which is important here since `d` is also a hex digit).

I am not going to block on this detail though.

Reply:
LXD does not really require the first letter to be alphabetic, adding d_ is just for the sake of indicating it is a device name. What the prefix stands for (device) is different from QEMU case, so I think it is ok to be a different letter. Removing dash can be applied here as well because it adds more digits in a way, the reason I did not do it is because it leads to more string manipulation which means more time cost, however, it might be small in this case.

To summarize, we can leave the first letter and length to be different because they are different in these two cases.